pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTElement.java

```diff
@@ -109,5 +109,18 @@ public boolean equals(Object o) {
     public int hashCode() {
         return ModelASTElement.class.hashCode();
     }
+
+    @Override
+    public Object clone()
```

Comment:
Formatting nitpick - I'd prefer consistent styling, i.e., `{` on the same line, `catch (CloneNotSupportedException e)` instead of `catch( CloneNotSupportedException e )`, etc.